refactor(chats): separate calls to getChats for each section

jrainville opened the pull request:
Fixes #9858

Uses the new status-go functions `getChannelGroups` (replacement of `getChats`) and `getChannelGroupByID`.

When we load the app, we call asyncly `getChannelGroups` to get the sections and show the navigation.

When a section is made active (so when we switch or on start, the last active section), we call `getChannelGroupByID` to get the chats, categories and members. I added a small placeholder loader to show we are loading them. It usually takes less than a second, but it's better to show something.

We only call `getChannelGroupByID` once per section, then it's "instant" to switch back.

Status-go PR: https://github.com/status-im/status-go/pull/3292

For QAs: test that all chats and sections still work. Check if history works too just in case.
Also test that unread markers and notifications still work for all channels, loaded or not.

jrainville commented:
I found an issue when you receive messages in a channel that wasn't loaded. You can review, but there will be some minor bug fixes added

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 71720743 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/tests-nim/job/PR-9914/1/) | 2023-03-15 20:24:56 | ~6 min | `tests-nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/tests-nim/job/PR-9914/1//consoleText) |

caybro commented on src/app_service/service/community/service.nim:
```diff
@@ -769,7 +769,8 @@ QtObject:
       updatedCommunity.settings = communitySettings
       self.communities[communityId] = updatedCommunity
-      self.chatService.loadChats()
+      # TODO this might be broken. Can we fetch only the community instead?
```
Still valid?

jrainville replied:
Yes. I'll fix it along with the other bug I found. Expect a small to medium commit coming in today

caybro commented on ui/app/AppLayouts/Chat/panels/ChatsLoadingPanel.qml:
```diff
@@ -0,0 +1,23 @@
+            StatusBaseText {
+                text: qsTr("Loading chats...")
+            }
+            LoadingAnimation {}
```
This might need `anchors.verticalCenter: parent.verticalCenter`

caybro reviewed (approved):
LGTM

alexjba commented:
I can confirm the app starts much faster. On my account I see about 6x improvement in startup time! 🎊

mprakhov reviewed (approved):
lgtm

jrainville commented:
Re-requesting review.
I added a commit that fixed the issue I found where getting a new message would create the chat before the chats were loaded, hence creating a double chat.

It was not simple, because we had a lot of convoluted code for historic reason. You'll find in my commit that I simplified a lot of the functions and instead of calling the controller/service to get data, I get it directly from the item/dto I have at hand, which is simpler and faster.

At first I tried to fix the issue by using `addChatIfDontExist` when creating the chat UI, so that if a chat already exists because of a previous message, we just ignore it, but it messed up the channel positions, because chats created from a signal don't have the `position` properties correctly set.
It still made me reuse the function, which at least made us remove duplication (for item creation for example)

I then made it so that we don't add the chats if we haven't loaded all chats before, but it created a problem that we didn't get the unread badge anymore. I fixed that by untangling the code and just using the count we get from the signal directly instead of recalculating from the model.

I then had to improve the `getChannelGroups` function further in status-go to provide me the initial count of each channel group for the initialization.

I'm squash the commits once you guys have reviewed.

caybro commented on src/app_service/service/chat/service.nim:
```diff
@@ -164,19 +169,39 @@ QtObject:
+  proc loadChannelGroupById*(self: Service, channelGroupId: string) =
+    try:
+      let response = status_chat.getChatsByChannelGroupId(channelGroupId)
+      echo "resposnere ", response
```
leftover?

caybro reviewed (approved):
Code LGTM, will give it a test tomorrow

igor-sirotin commented:
Fixed the `wait` label. We should probably remove the `statusq` one 🙂

anastasiyaig commented:
**tested:**

1:1 chats -> history, reactions, pins, replies, message deletion
group chats -> create, delete, fetch history, message actions, mentioning
community -> history, reactions, replies, mentioning, message deletion, links etc
also tested AC in terms of notifications and unread messages indicators. It works the same as master (or at least very close to it)
